### ISSUE TITLE
Failed to create Cognitive Service Account when qname_runtime_endpoint is not set

### ISSUE THREAD

**Issue by johejo** [terraform-providers/terraform-provider-azurerm#6343]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

v2.3.0 or newer

### Affected Resource(s)

* `azurerm_cognitive_account`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
resource "azurerm_cognitive_account" "cognitive_service" {
  name                = var.cognitive_service_name
  resource_group_name = var.resource_group_name
  location            = var.location
  kind                = "CognitiveServices"
  sku_name            = "S0"
}
```

### Expected Behavior

Successfully create cognitive account.

### Actual Behavior

```
azurerm_cognitive_account.cognitive_service: Creating...

Error: Error creating Cognitive Services Account ***: cognitiveservices.AccountsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="ApiPropertiesInvalid" Message="The given 'apiProperties' '{\r\n  \"qnaRuntimeEndpoint\": \"\"\r\n}' is invalid. Validation errors: Property 'qnaRuntimeEndpoint' has not been defined and the schema does not allow additional properties. Path 'apiProperties.qnaRuntimeEndpoint'."

  on main.tf line ***, in resource "azurerm_cognitive_account" "cognitive_service":
 ***: resource "azurerm_cognitive_account" "cognitive_service" {
```

https://github.com/terraform-providers/terraform-provider-azurerm/blob/c916aa6db4142add1760a9b39bc668eb8b1dfcde/azurerm/internal/services/cognitive/resource_arm_cognitive_account.go#L156-L166

Caused by #5778.
The 'qnaRuntimeEndpoint' type is `*string`.
If we assign a `""` using `utils.String` into 'qnaRuntimeEndpoint', we will get a `""` in json by marshaling.
We should use `d.GetOk("qname_runtime_endpoint")` and need to check if exists.
https://github.com/Azure/azure-sdk-for-go/blob/b7d185716507a401293b9f6c303f6e9798c4cb1b/services/cognitiveservices/mgmt/2017-04-18/cognitiveservices/models.go#L258-L268

### Steps to Reproduce

1. `terraform apply`

### References

* #5778

**Comment by devedse** [comment]
Hi all, I'm also running into this issue and was wondering what the status since the pull requests confuses me.

I assume that by writing tests @johejo figured out that when we don't pass the qna_runtime_endpoint, terraform doesn't include this in the API.

If that's the case, can we maybe try to simulate the API call directly and see if we get the same issue?

**Comment by katbyte** [comment]
@devedse - could you share you tf config?

**Comment by devedse** [comment]
@katbyte , this is my config:

```
resource "azurerm_cognitive_account" "cognitiveaccount" {
  name                = "${var.applicationName}-cognitive-${var.environment}"
  resource_group_name = azurerm_resource_group.resourcegroup.name
  location            = azurerm_resource_group.resourcegroup.location
  kind                = "CognitiveServices"

  qna_runtime_endpoint = "%s"

  sku_name = "S0"
}


resource "azurerm_key_vault_secret" "keyvaultsecret-cognitiveaccount-accesskey" {
  name         = "${var.applicationName}-cognitiveaccount-accesskey"
  value        = azurerm_cognitive_account.cognitiveaccount.primary_access_key
  key_vault_id = azurerm_key_vault.keyvault.id
}
```

**Comment by georgegil** [comment]
curious, is there a workaround? dummy URL for QNA maker?

**Comment by myliyifei** [comment]
I encountered the same error when creating the following "Bing.Search.v7" service, even though I added this (qna_runtime_endpoint  = null) parameter to the resource module.
  

>  kind        = "Bing.Search.v7"
     sku_name    = "F1"
     location    = "Global"

---
Error creating Cognitive Services Account "stock-news" (Resource Group "stock"): cognitiveservices.AccountsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="ApiPropertiesInvalid" Message="**The given 'apiProperties' '{\r\n  \"qnaRuntimeEndpoint\": \"\"\r\n}' is invalid. Validation errors: Property 'qnaRuntimeEndpoint' has not been defined and the schema does not allow additional properties. Path 'apiProperties.qnaRuntimeEndpoint'.**"
